plutus-ir/src/Language/PlutusIR/Compiler/Let.hs

```diff
@@ -43,7 +43,14 @@ compileLets kind t = getEnclosing >>= \p ->
 compileLet :: Compiling m e a => LetKind -> PIRTerm a -> DefT SharedName (Provenance a) m (PIRTerm a)
 compileLet kind = \case
     Let p r bs body -> withEnclosing (const $ LetBinding r p) $ case r of
-            NonRec -> lift $ foldM (compileNonRecBinding kind) body (reverse bs) -- this foldM is foldr
+            -- NOTE: 'foldM' for lists is left associative, but we need right associative for our case (every right let must be wrapped/scoped by its left let), example:
```

Comment:
Please follow the way the rest of the codebase does this: there's a particular format for notes and references to them that makes them easy to grep for!

i.e. top-level block comment beginning `Note [Name of Note]`, reference it with `See Note [Name of Note]`.